nifi-nar-bundles/nifi-deeplearning4j-bundle/nifi-deeplearning4j-processors/src/main/java/org/apache/nifi/processors/deeplearning4j/DeepLearning4JMultiLayerPredictor.java

```diff
@@ -0,0 +1,240 @@
+    static {
+        final Set<Relationship> tempRelationships = new HashSet<>();
+        tempRelationships.add(REL_SUCCESS);
+        tempRelationships.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(tempRelationships);
+        final List<PropertyDescriptor> tempDescriptors = new ArrayList<>();
+        tempDescriptors.add(MODEL_FILE);
+        tempDescriptors.add(RECORD_DIMENSIONS);
+        tempDescriptors.add(CHARSET);
+        tempDescriptors.add(FIELD_SEPARATOR);
+        tempDescriptors.add(RECORD_SEPARATOR);
+        propertyDescriptors = Collections.unmodifiableList(tempDescriptors);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    protected synchronized MultiLayerNetwork getModel(ProcessContext context) throws IOException {
+        if ( model == null ) {
+            String modelFile = context.getProperty(MODEL_FILE).evaluateAttributeExpressions().getValue();
+            getLogger().debug("Loading model from {}", new Object[] {modelFile});
+
+            long start = System.currentTimeMillis();
+            model = ModelSerializer.restoreMultiLayerNetwork(modelFile,false);
+            long end = System.currentTimeMillis();
+
+            getLogger().info("Time to load model " + (end-start) +  " ms");
+        }
+        return (MultiLayerNetwork)model;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        Charset charset = Charset.forName(context.getProperty(CHARSET).evaluateAttributeExpressions(flowFile).getValue());
+        if ( flowFile.getSize() == 0 ) {
+            String message = "FlowFile query is empty";
+            getLogger().error(message);
+            flowFile = session.putAttribute(flowFile, DEEPLEARNING4J_ERROR_MESSAGE, message);
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        String input = null;
+        try {
+            input = getFlowFileContents(session, charset, flowFile);
+            String fieldSeparator = context.getProperty(FIELD_SEPARATOR).evaluateAttributeExpressions(flowFile).getValue();
+            String recordSeparator = context.getProperty(RECORD_SEPARATOR).evaluateAttributeExpressions(flowFile).getValue();
+
+            int [] dimensions = getInputDimensions(context, charset, flowFile, fieldSeparator);
+
+            if ( getLogger().isDebugEnabled() )    {
+                getLogger().debug("Received input {} with dimensions {}", new Object[] { input, dimensions });
+            }
+
+            MultiLayerNetwork model = getModel(context);
```

Comment:
Would loading the model in an `@OnScheduled` function be better than waiting until the processor is triggered to load the model?

Reply:
@mans2singh Makes sense. I'm honestly not sure if `@OnScheduled` would be any better. Maybe someone else will comment. Thanks!